### PR TITLE
[VEN-105] Product states - pre-Orders support - drop available_on log…

### PR DESCRIPTION
**Pull request by kgorazd**
…ic - with both fields

Notes:
- Platform API supports both available_on, make_active_at.
- Storefront API supports on available_on currently.

### CODE REVIEW

File: core/app/finders/spree/products/find.rb [hunk 1]
```diff
@@ -215,7 +215,7 @@ def ordered(products)
         when 'name-z-a'
           products.order(name: :desc)
         when 'newest-first'
-          products.order(available_on: :desc)
+          products.order(make_active_at: :desc)
```
[the remark this answers is not in the document]
Comment (kgorazd) [reply core/app/finders/spree/products/find.rb]: done

File: core/app/models/concerns/spree/product_scopes.rb [hunk 2]
```diff
@@ -252,9 +252,9 @@ def self.with_currency(currency)
       search_scopes << :with_currency
 
       # Can't use add_search_scope for this as it needs a default argument
-      def self.available(available_on = nil, currency = nil)
```
[the remark this answers is not in the document]
Comment (kgorazd) [reply core/app/models/concerns/spree/product_scopes.rb]: done

File: core/db/migrate/20220103082046_add_status_and_make_active_at_to_spree_products.rb [hunk 3]
```diff
@@ -1,7 +1,9 @@
 class AddStatusAndMakeActiveAtToSpreeProducts < ActiveRecord::Migration[5.2]
   def change
     add_column :spree_products, :status, :string, null: false, default: 'draft'
+    add_column :spree_products, :make_active_at, :datetime
```
Comment (damianlegawiec) [inline core/db/migrate/20220103082046_add_status_and_make_active_at_to_spree_products.rb:4]: @kgorazd let's not modify existing migrations, let's add a new one please

File: core/lib/tasks/core.rake [hunk 4]
```diff
@@ -187,9 +187,9 @@ use rake db:load_file[/absolute/path/to/sample/filename.rb]}
 end
 
 namespace :core do
-  desc 'Set "active" status on draft products where available_on is in the past'
```
Comment (damianlegawiec) [inline core/lib/tasks/core.rake:190]: @kgorazd we also need to add in a migration adding make_active_at to set the active status for existing products that were available, and set status archived for discontinued products

File: core/db/migrate/20220103082046_add_status_and_make_active_at_to_spree_products.rb (hunk 3 above)
Comment (kgorazd) [reply core/db/migrate/20220103082046_add_status_and_make_active_at_to_spree_products.rb]: done

File: core/lib/tasks/core.rake (hunk 4 above)
Comment (kgorazd) [reply core/lib/tasks/core.rake:190]: done

File: core/db/migrate/20220117100333_add_make_active_at_to_spree_products.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class AddMakeActiveAtToSpreeProducts < ActiveRecord::Migration[5.2]
+  def change
+    add_column :spree_products, :make_active_at, :datetime
+    add_index :spree_products, :make_active_at
+    Spree::Product.where('make_active_at <= ?', Time.current).where(status: 'draft').update_all(status: 'active', updated_at: Time.current)
```
Comment (damianlegawiec) [inline core/db/migrate/20220117100333_add_make_active_at_to_spree_products.rb:5]: @kgorazd we need to check availability here, as make_active_at will be null for all

File: core/db/migrate/20220117100333_add_make_active_at_to_spree_products.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+class AddMakeActiveAtToSpreeProducts < ActiveRecord::Migration[5.2]
+  def change
+    add_column :spree_products, :make_active_at, :datetime
+    add_index :spree_products, :make_active_at
+    Spree::Product.where('available_on <= ?', Time.current).where(status: 'draft').update_all(status: 'active', updated_at: Time.current)
```
Comment (damianlegawiec) [inline core/db/migrate/20220117100333_add_make_active_at_to_spree_products.rb:5]: @kgorazd we need to check if these products aren't discontinued, it's possible to have both available on and discontinue on date set at the same time, I think we should use the same SQL query as the old `available` scope

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**